### ISSUE TITLE
Symbol not found: _OnigEncodingUTF8

### ISSUE THREAD

**Issue by rsalmei** [mwilliamson/jq.py#16]
Hello,

Please, could you help me?
I've installed with success, but can't import it...

Install:
```
ARCHFLAGS="-arch x86_64" pip install jq
Collecting jq
  Downloading jq-0.1.5.tar.gz
Installing collected packages: jq
  Running setup.py install for jq
Successfully installed jq-0.1.5
```

But in python repl:
```
>>> import jq
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ImportError: dlopen(/usr/local/lib/python2.7/site-packages/jq.so, 2): Symbol not found: _OnigEncodingUTF8
  Referenced from: /usr/local/lib/python2.7/site-packages/jq.so
  Expected in: flat namespace
 in /usr/local/lib/python2.7/site-packages/jq.so
```

Do you have any clue of what happened?
I'm in a mac os 10.10.5 Yosemite.
Thank you!!

**Comment by mwilliamson** [comment]
It looks like it's trying to use Oniguruma, but can't find it. Do you have Oniguruma installed?

What happens if you try to build jq itself? When you run `./configure`, does it find Oniguruma?

**Comment by rsalmei** [comment]
I don't have it installed, never even heard of Oniguruma...
Is it a runtime only dependency? How can I install it?

**Comment by mwilliamson** [comment]
I believe Oniguruma is the regex library that jq uses. As I understand it, it shouldn't be used if it's not installed, hence why I was asking whether it was found when running `./configure`. I don't have it installed, so your best bet is probably to take a look at how you install for plain jq.

**Comment by rsalmei** [comment]
Thanks. I didn't run `./configure`, just the pip install aforementioned.
But what you said got me thinking, if you do not have it (and it works on your machine), what could be wrong in mine?
I'd prefer not installing this Oniguruma too... 

I have jq installed via Homebrew, but since your python binding actually downloads the source code for jq and compiles it, I thought it would be self contained, even without jq it should work, right?

Any thoughts? Please! I'd really like to use this framework easily in python...
Thanks!

**Comment by mwilliamson** [comment]
> But what you said got me thinking, if you do not have it (and it works on your machine), what could be wrong in mine?

I don't run Mac OS, so it's a bit tricky to work out what the difference is.

> Thanks. I didn't run `./configure`, just the pip install aforementioned.

Could you run `./configure` on jq itself so that I can see the output? That way we can see if jq is trying to use Oniguruma.

**Comment by rsalmei** [comment]
Ohh, I see.

Ok, I'll try it.
I need to clone the jq source code repo, and just run `./configure` in it?

**Comment by mwilliamson** [comment]
Probably best to use the same tarball as the Python bindings: https://github.com/stedolan/jq/archive/jq-1.5.tar.gz

**Comment by rsalmei** [comment]
Ok, did with the same tarball as you suggested.

It runs with no problem!
I've even ran `make -j8` and called `jq` and it works.
See the log:
```
checking for a BSD-compatible install... /usr/local/bin/ginstall -c
checking whether build environment is sane... yes
checking for a thread-safe mkdir -p... /usr/local/bin/gmkdir -p
checking for gawk... no
checking for mawk... no
checking for nawk... no
checking for awk... awk
checking whether make sets $(MAKE)... yes
checking whether make supports nested variables... yes
checking whether make supports nested variables... (cached) yes
checking for style of include used by make... GNU
checking for gcc... gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables...
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether gcc accepts -g... yes
checking for gcc option to accept ISO C89... none needed
checking whether gcc understands -c and -o together... yes
checking dependency style of gcc... gcc3
checking for ar... ar
checking the archiver (ar) interface... ar
checking whether to enable maintainer-specific portions of Makefiles... no
checking for gcc... (cached) gcc
checking whether we are using the GNU C compiler... (cached) yes
checking whether gcc accepts -g... (cached) yes
checking for gcc option to accept ISO C89... (cached) none needed
checking whether gcc understands -c and -o together... (cached) yes
checking dependency style of gcc... (cached) gcc3
checking for gcc option to accept ISO C99... none needed
checking for gcc option to accept ISO Standard C... (cached) none needed
checking how to run the C preprocessor... gcc -E
checking for bison... bison -y
checking build system type... x86_64-apple-darwin14.5.0
checking host system type... x86_64-apple-darwin14.5.0
checking how to print strings... printf
checking for a sed that does not truncate output... /usr/local/bin/gsed
checking for grep that handles long lines and -e... /usr/bin/grep
checking for egrep... /usr/bin/grep -E
checking for fgrep... /usr/bin/grep -F
checking for ld used by gcc... /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ld
checking if the linker (/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ld) is GNU ld... no
checking for BSD- or MS-compatible name lister (nm)... /usr/bin/nm
checking the name lister (/usr/bin/nm) interface... BSD nm
checking whether ln -s works... yes
checking the maximum length of command line arguments... 196608
checking whether the shell understands some XSI constructs... yes
checking whether the shell understands "+="... yes
checking how to convert x86_64-apple-darwin14.5.0 file names to x86_64-apple-darwin14.5.0 format... func_convert_file_noop
checking how to convert x86_64-apple-darwin14.5.0 file names to toolchain format... func_convert_file_noop
checking for /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ld option to reload object files... -r
checking for objdump... no
checking how to recognize dependent libraries... pass_all
checking for dlltool... no
checking how to associate runtime and link libraries... printf %s\n
checking for archiver @FILE support... no
checking for strip... strip
checking for ranlib... ranlib
checking command to parse /usr/bin/nm output from gcc object... ok
checking for sysroot... no
checking for mt... no
checking if : is a manifest tool... no
checking for dsymutil... dsymutil
checking for nmedit... nmedit
checking for lipo... lipo
checking for otool... otool
checking for otool64... no
checking for -single_module linker flag... yes
checking for -exported_symbols_list linker flag... yes
checking for -force_load linker flag... yes
checking for ANSI C header files... yes
checking for sys/types.h... yes
checking for sys/stat.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for memory.h... yes
checking for strings.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for unistd.h... yes
checking for dlfcn.h... yes
checking for objdir... .libs
checking if gcc supports -fno-rtti -fno-exceptions... yes
checking for gcc option to produce PIC... -fno-common -DPIC
checking if gcc PIC flag -fno-common -DPIC works... yes
checking if gcc static flag -static works... no
checking if gcc supports -c -o file.o... yes
checking if gcc supports -c -o file.o... (cached) yes
checking whether the gcc linker (/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ld) supports shared libraries... yes
checking dynamic linker characteristics... darwin14.5.0 dyld
checking how to hardcode library paths into programs... immediate
checking whether stripping libraries is possible... yes
checking if libtool supports shared libraries... yes
checking whether to build shared libraries... yes
checking whether to build static libraries... yes
checking oniguruma.h usability... yes
checking oniguruma.h presence... yes
checking for oniguruma.h... yes
checking for onig_version in -lonig... yes
checking for valgrind... no
configure: WARNING: valgrind is required to test jq.
checking for memmem... yes
checking for mkstemp... yes
checking for bundle... bundle
checking for Ruby dependencies... configure: WARNING: The following gems are missing
 * rake (10.4.2)
 * json (1.8.3)
 * minitest (5.8.0)
 * thread_safe (0.3.5)
 * tzinfo (1.2.2)
 * activesupport (4.2.3)
 * addressable (2.3.8)
 * builder (3.2.2)
 * launchy (2.4.3)
 * liquid (3.0.6)
 * maruku (0.7.2)
 * rack (1.6.4)
 * sass (3.4.16)
 * rack-protection (1.5.3)
 * tilt (2.0.1)
 * sinatra (1.4.6)
 * watch (0.1.0)
 * yui-compressor (0.12.0)
 * bonsai (1.4.9)
 * hpricot (0.8.6)
 * mustache (0.99.8)
 * rdiscount (2.1.8)
 * ronn (0.7.3)
Install missing gems with `bundle install`
*****************************************************************
*  Ruby dependencies for building jq documentation not found.   *
*  You can still build, install and hack on jq, but the manpage *
*  will not be rebuilt and some of the tests won't run.         *
*  See docs/README.md for how to install the docs dependencies. *
*****************************************************************
no
checking for size_t... yes
checking for working alloca.h... yes
checking for alloca... yes
checking for isatty... yes
checking for _isatty... no
checking for strptime... yes
checking for strftime... yes
checking for timegm... yes
checking for gmtime_r... yes
checking for gmtime... yes
checking for gettimeofday... yes
checking for struct tm.tm_gmtoff... yes
checking for struct tm.__tm_gmtoff... no
checking for acos... yes
checking for acosh... yes
checking for asin... yes
checking for asinh... yes
checking for atan... yes
checking for atanh... yes
checking for cbrt... yes
checking for cos... yes
checking for cosh... yes
checking for exp2... yes
checking for exp... yes
checking for floor... yes
checking for j0... yes
checking for j1... yes
checking for log10... yes
checking for log2... yes
checking for log... yes
checking for sin... yes
checking for sinh... yes
checking for sqrt... yes
checking for tan... yes
checking for tanh... yes
checking for tgamma... yes
checking for y0... yes
checking for y1... yes
checking for pow... yes
checking for atan2... yes
checking for hypot... yes
checking for remainder... yes
checking for thread-local storage... yes
checking whether byte ordering is bigendian... no
checking that generated files are newer than configure... done
configure: creating ./config.status
config.status: creating Makefile
config.status: executing depfiles commands
config.status: executing libtool commands
```

**Comment by mwilliamson** [comment]
Looks like you do have Oniguruma installed (or at least jq thinks you do) since the output shows it being found:

```
checking oniguruma.h usability... yes
checking oniguruma.h presence... yes
checking for oniguruma.h... yes
checking for onig_version in -lonig... yes
```

But it seems like it can't find the dynamic libraries at runtime. I don't know about how Mac OS works with respect to installation of libraries, so probably can't help much in that respect.

**Comment by rsalmei** [comment]
Well, thats a bummer.
Thank you for your effort.
Can we let this issue open, to see if some day anyone can figure it out?
Thanks!

**Comment by rsalmei** [comment]
Please @mwilliamson, could you see this log trying to install on a CentOS 6.5?
Initially it failed because it required autoconf 2.64, and it was installed 2.63.
I've updated it to 2.69 and now I get this (don't know linux as much as macos):

```
[vagrant@devhost vagrant]$ pip install jq
You are using pip version 6.1.1, however version 7.1.2 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.
Collecting jq
  Using cached jq-0.1.5.tar.gz
Installing collected packages: jq
  Running setup.py install for jq
    Complete output from command /home/vagrant/.pyenv/versions/2.7.10/bin/python2.7 -c "import setuptools, tokenize;__file__='/tmp/pip-build-qfs1GL/jq/setup.py';exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\r\n', '\n'), __file__, 'exec'))" install --record /tmp/pip-O8oGFC-record/install-record.txt --single-version-externally-managed --compile:
    running install
    running build
    running build_ext
    configure.ac:14: warning: macro `AM_PROG_AR' not found in library
    fatal: Not a git repository (or any of the parent directories): .git
    fatal: Not a git repository (or any of the parent directories): .git
    libtoolize: `config/ltmain.sh' is newer: use `--force' to overwrite
    libtoolize: `config/m4/libtool.m4' is newer: use `--force' to overwrite
    libtoolize: `config/m4/ltoptions.m4' is newer: use `--force' to overwrite
    libtoolize: `config/m4/ltversion.m4' is newer: use `--force' to overwrite
    libtoolize: `config/m4/lt~obsolete.m4' is newer: use `--force' to overwrite
    configure.ac:14: warning: macro `AM_PROG_AR' not found in library
    configure.ac:14: error: possibly undefined macro: AM_PROG_AR
          If this token and others are legitimate, please use m4_pattern_allow.
          See the Autoconf documentation.
    autoreconf: /usr/local/bin/autoconf failed with exit status: 1
    Executing: autoreconf -i
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-build-qfs1GL/jq/setup.py", line 87, in <module>
        'Programming Language :: Python :: 3.4',
      File "/home/vagrant/.pyenv/versions/2.7.10/lib/python2.7/distutils/core.py", line 151, in setup
        dist.run_commands()
      File "/home/vagrant/.pyenv/versions/2.7.10/lib/python2.7/distutils/dist.py", line 953, in run_commands
        self.run_command(cmd)
      File "/home/vagrant/.pyenv/versions/2.7.10/lib/python2.7/distutils/dist.py", line 972, in run_command
        cmd_obj.run()
      File "/home/vagrant/.pyenv/versions/2.7.10/lib/python2.7/site-packages/setuptools/command/install.py", line 61, in run
        return orig.install.run(self)
      File "/home/vagrant/.pyenv/versions/2.7.10/lib/python2.7/distutils/command/install.py", line 563, in run
        self.run_command('build')
      File "/home/vagrant/.pyenv/versions/2.7.10/lib/python2.7/distutils/cmd.py", line 326, in run_command
        self.distribution.run_command(command)
      File "/home/vagrant/.pyenv/versions/2.7.10/lib/python2.7/distutils/dist.py", line 972, in run_command
        cmd_obj.run()
      File "/home/vagrant/.pyenv/versions/2.7.10/lib/python2.7/distutils/command/build.py", line 127, in run
        self.run_command(cmd_name)
      File "/home/vagrant/.pyenv/versions/2.7.10/lib/python2.7/distutils/cmd.py", line 326, in run_command
        self.distribution.run_command(command)
      File "/home/vagrant/.pyenv/versions/2.7.10/lib/python2.7/distutils/dist.py", line 972, in run_command
        cmd_obj.run()
      File "/tmp/pip-build-qfs1GL/jq/setup.py", line 52, in run
        command(["autoreconf", "-i"])
      File "/tmp/pip-build-qfs1GL/jq/setup.py", line 46, in command
        subprocess.check_call(args, cwd=jq_lib_dir)
      File "/home/vagrant/.pyenv/versions/2.7.10/lib/python2.7/subprocess.py", line 540, in check_call
        raise CalledProcessError(retcode, cmd)
    subprocess.CalledProcessError: Command '['autoreconf', '-i']' returned non-zero exit status 1

    ----------------------------------------
    Command "/home/vagrant/.pyenv/versions/2.7.10/bin/python2.7 -c "import setuptools, tokenize;__file__='/tmp/pip-build-qfs1GL/jq/setup.py';exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\r\n', '\n'), __file__, 'exec'))" install --record /tmp/pip-O8oGFC-record/install-record.txt --single-version-externally-managed --compile" failed with error code 1 in /tmp/pip-build-qfs1GL/jq
[vagrant@devhost vagrant]$
```

**Comment by rsalmei** [comment]
Never mind, updated automake to 1.14 and it has worked!
So, we're back to original problem, how to make it function on my mac yosemite...
Thanks!

**Comment by cooltrane** [comment]
seems to be not an Mac OSX only problem.

I am having very similar problems under archlinux python3.5.

I tried : 
- standard pip install jq
- ./configure && make && python3 setup.py install with the earlier mentionned tar.gz

And I keep having the same symbol error on ongiuruma (installed on my system) : 
```python ⓔ  35-jq  In [1]  import jq
---------------------------------------------------------------------------
ImportError                               Traceback (most recent call last)
<ipython-input-1-e1cfd80b6ce2> in <module>()
----> 1 import jq

ImportError: /home/thomas/35-jq/lib/python3.5/site-packages/jq.cpython-35m-x86_64-linux-gnu.so: undefined symbol: OnigSyntaxPerl_NG
```

**Comment by mwilliamson** [comment]
When you run `./configure`, does it find Oniguruma?

**Comment by cooltrane** [comment]
yep it does ->
[~/jq-jq-1.5]> ./configure
...
checking oniguruma.h usability... yes
checking oniguruma.h presence... yes
checking for oniguruma.h... yes
checking for onig_version in -lonig... yes
...

**Comment by cooltrane** [comment]
Also, when using pip install I have a traceback (that I didn't notice yesterday).

I am doing my tests with a python3.5 virtualenv by the way.

  Traceback (most recent call last):
    File "<string>", line 1, in <module>
    File "/tmp/pip-build-trz4ofzn/jq/setup.py", line 87, in <module>
      'Programming Language :: Python :: 3.4',
    File "/usr/lib64/python3.5/distutils/core.py", line 148, in setup
      dist.run_commands()
    File "/usr/lib64/python3.5/distutils/dist.py", line 955, in run_commands
      self.run_command(cmd)
    File "/usr/lib64/python3.5/distutils/dist.py", line 974, in run_command
      cmd_obj.run()
    File "/home/thomas/jqtest/lib/python3.5/site-packages/wheel/bdist_wheel.py", line 213, in run
      archive_basename = self.get_archive_basename()
    File "/home/thomas/jqtest/lib/python3.5/site-packages/wheel/bdist_wheel.py", line 161, in get_archive_basename
      impl_tag, abi_tag, plat_tag = self.get_tag()
    File "/home/thomas/jqtest/lib/python3.5/site-packages/wheel/bdist_wheel.py", line 155, in get_tag
      assert tag == supported_tags[0]
  AssertionError

do not see why it would be related, but just in case.

**Comment by cooltrane** [comment]
I just made a test with python2.7 to be sure:

do not have the above traceback for the pip install.

but still : 
```
in [7] > sys.version
Out[7] > '2.7.10 (default, Sep  7 2015, 13:51:49) \n[GCC 5.2.0]'

In [8] > import jq
---------------------------------------------------------------------------
ImportError                               Traceback (most recent call last)
<ipython-input-8-e1cfd80b6ce2> in <module>()
----> 1 import jq

ImportError: /home/thomas/py2jq/lib/python2.7/site-packages/jq.so: undefined symbol: OnigSyntaxPerl_NG
```

**Comment by mwilliamson** [comment]
Can you successfully use jq with the regex features?

**Comment by cooltrane** [comment]
mwilliamson: what do you mean ?
The jq executable ? yep I can, been using it for a while.
```
┌[thomas@archlaptgerbier] [/dev/pts/2] [3]
└[~]> jq -n '("test", "TEst", "teST", "TEST") | test( "(?i)te(?-i)st" )'
true
true
false
false
```

**Comment by mwilliamson** [comment]
> The jq executable ? yep I can, been using it for a while.

Did you compile from source or use the pre-built binary?

**Comment by mwilliamson** [comment]
To expand a little: if you compiled from source and it's using the dynamic libraries, then I'm not entirely sure why the binary can find the libs but the Python bindings can't. Otherwise, it just seems like the Oniguruma dynamic libraries are in a directory that the dynamic linker isn't configured to use. There's [a decent explanation on a jq issue](https://github.com/stedolan/jq/issues/613#issuecomment-72905968):

> You'll probably need to do two things (I know I did) once installing onigurama.
>
>  * Add the correct path to your libraries path(s)
>  * reload this setting
>
> If you run ldd against your knackered binary you'll see that libonig.so.2 is not found.
>
> if you create a file at, for example, /etc/ld.so.conf.d/onig.conf with the directory libonig has been installed to:
> 
>     $ cat /etc/ld.so.conf.d/onig.conf
>     /usr/local/lib
>
> Then run ldconfig and dynamic linking will work.

**Comment by glyph** [comment]
I had a similar issue after `pip install --user jq` and then again in a virtualenv.  Installed jq itself with `brew install jq`.

**Comment by glyph** [comment]
```
ProductName:	Mac OS X
ProductVersion:	10.11.1
BuildVersion:	15B42
```

**Comment by mwilliamson** [comment]
When you run `./configure`, I assume it finds the Oniguruma header files? Are you sure the Oniguruma dynamic libraries are on the library path? You can check by running `ldd` (or possibly `otool`) on the binary jq library that gets generated as part of the bindings.

**Comment by glyph** [comment]
I didn't run `./configure`, just `pip`.  `jq` was already installed by brew so I didn't run `./configure` there either.  `otool` tells me that the `jq.so` doesn't link Oniguruma at all, just libSystem:

```
/Users/glyph/Library/Python/2.7/lib/python/site-packages/jq.so:
	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1226.10.1)
```

**Comment by mwilliamson** [comment]
Would you mind trying to install from the git branch `static-jq` and see if that makes any difference?

**Comment by glyph** [comment]
That branch doesn't seem to build

```console
$ pip install --upgrade 'git+https://github.com/mwilliamson/jq.py@static-jq'
Collecting git+https://github.com/mwilliamson/jq.py@static-jq
  Cloning https://github.com/mwilliamson/jq.py (to static-jq) to /var/folders/qd/1kj94mj96sv9bq60xgb3hzfh0000gp/T/pip-Mi39IB-build
Installing collected packages: jq
  Found existing installation: jq 0.1.5
    Uninstalling jq-0.1.5:
      Successfully uninstalled jq-0.1.5
  Running setup.py install for jq
    Complete output from command /usr/local/opt/python/bin/python2.7 -c "import setuptools, tokenize;__file__='/var/folders/qd/1kj94mj96sv9bq60xgb3hzfh0000gp/T/pip-Mi39IB-build/setup.py';exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\r\n', '\n'), __file__, 'exec'))" install --record /var/folders/qd/1kj94mj96sv9bq60xgb3hzfh0000gp/T/pip-JWOIm8-record/install-record.txt --single-version-externally-managed --compile --user --prefix=:
    running install
    running build
    running build_ext
    fatal: No names found, cannot describe anything.
    fatal: No names found, cannot describe anything.
    fatal: No names found, cannot describe anything.
    glibtoolize: putting auxiliary files in AC_CONFIG_AUX_DIR, 'config'.
    glibtoolize: copying file 'config/ltmain.sh'
    glibtoolize: putting macros in AC_CONFIG_MACRO_DIRS, 'config/m4'.
    glibtoolize: copying file 'config/m4/libtool.m4'
    glibtoolize: copying file 'config/m4/ltoptions.m4'
    glibtoolize: copying file 'config/m4/ltversion.m4'
    fatal: No names found, cannot describe anything.
    fatal: No names found, cannot describe anything.
    fatal: No names found, cannot describe anything.
    fatal: No names found, cannot describe anything.
    checking for a BSD-compatible install... /usr/bin/install -c
    checking whether build environment is sane... yes
    checking for a thread-safe mkdir -p... config/install-sh -c -d
    checking for gawk... no
    checking for mawk... no
    checking for nawk... no
    checking for awk... awk
    checking whether make sets $(MAKE)... yes
    checking whether make supports nested variables... yes
    checking whether make supports nested variables... (cached) yes
    checking for style of include used by make... GNU
    checking for gcc... gcc
    checking whether the C compiler works... yes
    checking for C compiler default output file name... a.out
    checking for suffix of executables...
    checking whether we are cross compiling... no
    checking for suffix of object files... o
    checking whether we are using the GNU C compiler... yes
    checking whether gcc accepts -g... yes
    checking for gcc option to accept ISO C89... none needed
    checking whether gcc understands -c and -o together... yes
    checking dependency style of gcc... gcc3
    checking for ar... ar
    checking the archiver (ar) interface... ar
    checking whether to enable maintainer-specific portions of Makefiles... no
    checking for gcc... (cached) gcc
    checking whether we are using the GNU C compiler... (cached) yes
    checking whether gcc accepts -g... (cached) yes
    checking for gcc option to accept ISO C89... (cached) none needed
    checking whether gcc understands -c and -o together... (cached) yes
    checking dependency style of gcc... (cached) gcc3
    checking for gcc option to accept ISO C99... none needed
    checking for gcc option to accept ISO Standard C... (cached) none needed
    checking how to run the C preprocessor... gcc -E
    checking for bison... bison -y
    checking build system type... x86_64-apple-darwin15.2.0
    checking host system type... x86_64-apple-darwin15.2.0
    checking how to print strings... printf
    checking for a sed that does not truncate output... /usr/bin/sed
    checking for grep that handles long lines and -e... /usr/bin/grep
    checking for egrep... /usr/bin/grep -E
    checking for fgrep... /usr/bin/grep -F
    checking for ld used by gcc... /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ld
    checking if the linker (/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ld) is GNU ld... no
    checking for BSD- or MS-compatible name lister (nm)... /usr/bin/nm
    checking the name lister (/usr/bin/nm) interface... BSD nm
    checking whether ln -s works... yes
    checking the maximum length of command line arguments... 196608
    checking how to convert x86_64-apple-darwin15.2.0 file names to x86_64-apple-darwin15.2.0 format... func_convert_file_noop
    checking how to convert x86_64-apple-darwin15.2.0 file names to toolchain format... func_convert_file_noop
    checking for /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ld option to reload object files... -r
    checking for objdump... no
    checking how to recognize dependent libraries... pass_all
    checking for dlltool... no
    checking how to associate runtime and link libraries... printf %s\n
    checking for archiver @FILE support... no
    checking for strip... strip
    checking for ranlib... ranlib
    checking command to parse /usr/bin/nm output from gcc object... ok
    checking for sysroot... no
    checking for a working dd... /bin/dd
    checking how to truncate binary pipes... /bin/dd bs=4096 count=1
    checking for mt... no
    checking if : is a manifest tool... no
    checking for dsymutil... dsymutil
    checking for nmedit... nmedit
    checking for lipo... lipo
    checking for otool... otool
    checking for otool64... no
    checking for -single_module linker flag... yes
    checking for -exported_symbols_list linker flag... yes
    checking for -force_load linker flag... yes
    checking for ANSI C header files... yes
    checking for sys/types.h... yes
    checking for sys/stat.h... yes
    checking for stdlib.h... yes
    checking for string.h... yes
    checking for memory.h... yes
    checking for strings.h... yes
    checking for inttypes.h... yes
    checking for stdint.h... yes
    checking for unistd.h... yes
    checking for dlfcn.h... yes
    checking for objdir... .libs
    checking if gcc supports -fno-rtti -fno-exceptions... yes
    checking for gcc option to produce PIC... -fno-common -DPIC
    checking if gcc PIC flag -fno-common -DPIC works... yes
    checking if gcc static flag -static works... no
    checking if gcc supports -c -o file.o... yes
    checking if gcc supports -c -o file.o... (cached) yes
    checking whether the gcc linker (/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ld) supports shared libraries... yes
    checking dynamic linker characteristics... darwin15.2.0 dyld
    checking how to hardcode library paths into programs... immediate
    checking whether stripping libraries is possible... yes
    checking if libtool supports shared libraries... yes
    checking whether to build shared libraries... yes
    checking whether to build static libraries... yes
    checking oniguruma.h usability... yes
    checking oniguruma.h presence... yes
    checking for oniguruma.h... yes
    checking for onig_version in -lonig... yes
    checking for valgrind... no
    configure: WARNING: valgrind is required to test jq.
    checking for memmem... yes
    checking for mkstemp... yes
    checking for bundle... bundle
    checking for Ruby dependencies... configure: WARNING:
    *****************************************************************
    *  Ruby dependencies for building jq documentation not found.   *
    *  You can still build, install and hack on jq, but the manpage *
    *  will not be rebuilt and some of the tests won't run.         *
    *  See docs/README.md for how to install the docs dependencies. *
    *****************************************************************
    no
    checking for size_t... yes
    checking for working alloca.h... yes
    checking for alloca... yes
    checking for isatty... yes
    checking for _isatty... no
    checking for strptime... yes
    checking for strftime... yes
    checking for timegm... yes
    checking for gmtime_r... yes
    checking for gmtime... yes
    checking for gettimeofday... yes
    checking for struct tm.tm_gmtoff... yes
    checking for struct tm.__tm_gmtoff... no
    checking for acos... yes
    checking for acosh... yes
    checking for asin... yes
    checking for asinh... yes
    checking for atan... yes
    checking for atanh... yes
    checking for cbrt... yes
    checking for cos... yes
    checking for cosh... yes
    checking for exp2... yes
    checking for exp... yes
    checking for floor... yes
    checking for j0... yes
    checking for j1... yes
    checking for log10... yes
    checking for log2... yes
    checking for log... yes
    checking for sin... yes
    checking for sinh... yes
    checking for sqrt... yes
    checking for tan... yes
    checking for tanh... yes
    checking for tgamma... yes
    checking for y0... yes
    checking for y1... yes
    checking for pow... yes
    checking for atan2... yes
    checking for hypot... yes
    checking for remainder... yes
    checking for thread-local storage... yes
    checking whether byte ordering is bigendian... no
    checking that generated files are newer than configure... done
    configure: creating ./config.status
    config.status: creating Makefile
    config.status: executing depfiles commands
    config.status: executing libtool commands
    /Applications/Xcode.app/Contents/Developer/usr/bin/make  all-am
      CC       locfile.lo
      CC       bytecode.lo
      CC       compile.lo
      CC       execute.lo
      CC       builtin.lo
      CC       jv.lo
      CC       jv_parse.lo
      CC       jv_print.lo
      CC       jv_dtoa.lo
      CC       jv_unicode.lo
      CC       jv_aux.lo
      CC       jv_file.lo
      CC       jv_alloc.lo
      CC       jq_test.lo
      CC       util.lo
      CC       linker.lo
      CC       parser.lo
      CC       lexer.lo
      CCLD     libjq.la
    libtool: warning: complete static linking is impossible in this configuration
      GEN      version.h
      CC       main.o
      CCLD     jq
      GEN      jq.1
    Executing: autoreconf -i
    Executing: ./configure CFLAGS=-fPIC --disable-maintainer-mode
    Executing: make LDFLAGS=-all-static
    building 'jq' extension
    creating build
    creating build/temp.macosx-10.11-x86_64-2.7
    clang -fno-strict-aliasing -fno-common -dynamic -g -O2 -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -I/var/folders/qd/1kj94mj96sv9bq60xgb3hzfh0000gp/T/pip-Mi39IB-build/jq-jq-1.5 -I/usr/local/include -I/usr/local/opt/openssl/include -I/usr/local/opt/sqlite/include -I/usr/local/Cellar/python/2.7.11/Frameworks/Python.framework/Versions/2.7/include/python2.7 -c jq.c -o build/temp.macosx-10.11-x86_64-2.7/jq.o
    clang: error: no such file or directory: 'jq.c'
    clang: error: no input files
    error: command 'clang' failed with exit status 1
    
    ----------------------------------------
  Rolling back uninstall of jq
Command "/usr/local/opt/python/bin/python2.7 -c "import setuptools, tokenize;__file__='/var/folders/qd/1kj94mj96sv9bq60xgb3hzfh0000gp/T/pip-Mi39IB-build/setup.py';exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\r\n', '\n'), __file__, 'exec'))" install --record /var/folders/qd/1kj94mj96sv9bq60xgb3hzfh0000gp/T/pip-JWOIm8-record/install-record.txt --single-version-externally-managed --compile --user --prefix=" failed with error code 1 in /var/folders/qd/1kj94mj96sv9bq60xgb3hzfh0000gp/T/pip-Mi39IB-build
```

**Comment by mwilliamson** [comment]
Is Cython installed?

**Comment by glyph** [comment]
Previously, no, but I get the exact same error after installing it.

**Comment by mwilliamson** [comment]
Ah, yes, installing Cython won't have changed anything since `setup.py` doesn't generate the `.c` file.

Could you clone the repository and try running `make test` on that branch?

**Comment by mwilliamson** [comment]
The branch `oniguruma` includes explicit support for Oniguruma. If you've been having this issue, could you try it out and let me know whether it works or not? It's not intended to be directly installable from git, so you'll need to clone the repoitory, checkout the branch, install cython and run `cython jq.pyx` before trying to install it.

**Comment by mwilliamson** [comment]
The changes from said branch have been released as 0.1.6, which should fix this issue. Let me know if that's not the case.
